packages/modules/vehicles/smarteq/api.py

```diff
@@ -225,25 +386,48 @@ async def _fetch_soc(self,
 
         try:
             soc = -1
-            if 'access_token' in self.Tokens:
+            range = 0.0
+            if 'refresh_token' in self.store['Tokens']:
+                self.store['Tokens'] = self.reconnect()
+            if 'access_token' in self.store['Tokens']:
                 soc, range = self.get_status(self.vin)
-                if soc > -1:
-                    self.log.info("get_status success- valid token")
+                if soc > 0:
+                    self.log.debug("fetch_soc: 1st attempt successful")
+                else:
+                    self.log.debug("fetch_soc: 1st attempt failed - soc=" + str(soc))
 
             if soc == -1:
-                self.log.error("get_status failed, soc= " + str(soc))
-                self.log.error("get_status failed, reconnecting ...")
-                self.resume = self.get_resume()
-                self.Tokens = self.reconnect()
-                soc, range = self.get_status(self.vin)
+                self.log.debug("fetch_soc: (re)connecting ...")
+                self.store['Tokens'] = self.reconnect()
+                if 'access_token' in self.store['Tokens']:
+                    soc, range = self.get_status(self.vin)
+                    if soc > 0:
+                        self.log.debug("fetch_soc: 2nd attempt successful")
+                    else:
+                        self.log.warning("fetch_soc: 2nd attempt failed - soc=" + str(soc))
+                        range = 0.0
+                else:
+                    self.log.error("fetch_soc: (re-)connect failed")
+                    soc = 0
+                    range = 0.0
             elif soc == -2:
                 self.log.error("get_status failed, verify VIN ...")
+                soc = 0
+                range = 0.0
 
         except Exception as e:
-            self.log.exception("get_status failed, reconnecting ..." + str(e))
-            self.resume = self.get_resume()
-            self.Tokens = self.reconnect()
-            soc, range = self.get_status(self.vin)
+            self.log.error("fetch_soc: exception, (re-)connecting ..." + str(e))
+            self.store['Tokens'] = self.reconnect()
+            if 'access_token' in self.store['Tokens']:
+                soc, range = self.get_status(self.vin)
+        self.log.info(" SOC/Range: " + str(soc) + '%/' + str(range) +
+                      '@' + self.soc_ts +
+                      ', Method: ' + self.method)
+
+        if self.store['Tokens'] != self.oldTokens:
+            self.log.debug("reconnect: tokens changed, store token file")
+
+        self.write_store()
```

Comment:
```suggestion
        if self.store['Tokens'] != self.oldTokens:
            self.log.debug("reconnect: tokens changed, store token file")
            self.write_store()
```
Die Logmeldung passt nicht, wenn die Tokens immer gespeichert werden. Vermutlich sollen sie aber auch nur bei Änderung gespeichert werden.